File: booktheturf_backend/src/main/java/com/dev/booktheturf/entity/Turf.java
package com.dev.booktheturf.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "turfs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Turf {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String location;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 2000)
    private String imageUrl;

    private String sportsType;

    private Double rating;

    private Double pricePerHour;

    private String contactNumber;
}

File: booktheturf_frontend/src/types/Turf.ts
export interface Turf {
  id?: number;
  name: string;
  location: string;
  description: string;
  pricePerHour: number;
  contactNumber: string;
  imageUrl: string;
  sportsType: string;
  rating: number;
  ratingCount?: number;
  distance?: string;
  area?: string;
  address?: string;
  amenities?: string[];
  images?: string[];
  isBookable?: boolean;
  sportsTypes?: string[];
}
